nbqa/__main__.py

```diff
@@ -105,7 +138,9 @@ def _get_all_notebooks(root_dirs: List[str]) -> Iterator[Path]:
     Iterator
         All Jupyter Notebooks found in all passed directories/notebooks.
     """
-    return (j for i in root_dirs for j in _get_notebooks(i))
+    return _filter_by_include_exclude(
+        (j for i in root_dirs for j in _get_notebooks(i)), files, exclude
```

Comment:
Use `nbqa-files` patterns in `_get_notebooks` function as part of `Path.rglob()` and use `--nbqa-exclude` in `_filter_by_exclude`.

Reply:
If we use regular expressions then I don't think this'll work - plus, like this, we can copy [filter_by_exclude](https://github.com/pre-commit/pre-commit/blob/e836e554891beba0e8e51092116559004772bf68/pre_commit/commands/run.py#L61) from pre-commit, which we know works

Reply:
Btw. don't forget to give props to the `pre-commit` code 😉 
E.g. inspired by <permalink_to_pre-commit_code>

Reply:
you're right, I gotta do that!